src/backend/distributed/commands/common.c

```diff
@@ -0,0 +1,284 @@
+/*-------------------------------------------------------------------------
+ *
+ * common.c
+ *
+ *    Most of the object propagation code consists of mostly the same
+ *    operations, varying slightly in parameters passed around. This
+ *    file contains most of the reusable logic in object propagation.
+ *
+ * Copyright (c) Citus Data, Inc.
+ *
+ *-------------------------------------------------------------------------
+ */
+
+#include "postgres.h"
+
+#include "catalog/objectaddress.h"
+#include "nodes/parsenodes.h"
+#include "tcop/utility.h"
+
+#include "distributed/commands.h"
+#include "distributed/commands/utility_hook.h"
+#include "distributed/deparser.h"
+#include "distributed/listutils.h"
+#include "distributed/metadata_sync.h"
+#include "distributed/metadata/dependency.h"
+#include "distributed/metadata/distobject.h"
+#include "distributed/multi_executor.h"
+#include "distributed/worker_transaction.h"
+
+List *
+PreprocessCreateDistributedObjectFromCatalogStmt(Node *node, const char *queryString,
+												 ProcessUtilityContext
+												 processUtilityContext)
+{
+	QualifyTreeNode((Node *) node);
```

Comment:
Would it be possible to move this call into ProcessUtilityInternal?

Reply:
It is possible, this also allows to remove many of its invocations in other preprocessing. The side-effect will be that, for statements that are _only_ impacting local objects we also qualify the objects. This should not cause issues, but if it does it is not easy to work around.

Do we still have automated tests where we run all postgres tests with citus installed? This could have some side effects in error messages we see there.

Reply:
#5933

Reply:
We do, but we do not create the extension (since that causes some tests to fail).

We should probably add an option to create the extension when running them locally. Example of enabling it:
https://github.com/citusdata/citus/commit/740ab078aa9e1c0c0f7eef52b6508a3a8fe2b1f6

Reply:
> I reverted my changes here,

makes sense